packages/vitepress-plugin-og-image-generator/src/index.ts

```diff
@@ -0,0 +1,97 @@
+import { fileURLToPath } from 'node:url'
+import { dirname, relative, resolve, sep } from 'node:path'
+import fs from 'fs-extra'
+import fg from 'fast-glob'
+import { Resvg } from '@resvg/resvg-js'
+import type { ArticleTree, Doc } from './types/metadata'
+import { removeEmoji } from './utils'
+
+const DIR_ROOT = resolve(dirname(fileURLToPath(import.meta.url)), '..')
+const dist = '.vitepress/dist/'
+const ogSvg = fs.readFileSync(resolve(DIR_ROOT, './public/og-template.svg'), 'utf-8')
+
+async function generateSVG(article: ArticleTree, output: string) {
+  const lines = removeEmoji(article.text).trim().replace(/(?![^\n]{1,17}$)([^\n]{1,17})\s/g, '$1\n').split('\n')
+  lines.forEach((val, i) => {
+    if (val.length > 17) {
+      lines[i] = val.slice(0, 17)
+      lines[i + 1] = `${val.slice(17)}${lines[i + 1] || ''}`
+    }
+    lines[i] = lines[i].trim()
+  })
+
+  const category = article.category ? removeEmoji(article.category).trim() : ''
+
+  const data = {
+    name1: lines[0] || '',
+    name2: lines[1] || '',
+    name3: `${lines[2] || ''}${lines[3] ? '...' : ''}`,
+    category,
+  }
+
+  const svg = ogSvg.replace(/\{\{([^}]+)}}/g, (_, name) => {
+    if (!name)
+      return ''
+    if (typeof name !== 'string')
+      return ''
+    if (!(name in data))
+      return ''
+
+    const nameKeyOf = name as keyof typeof data
+    return data[nameKeyOf]
+  })
+
+  const resvg = new Resvg(svg, {
+    fitTo: { mode: 'width', value: 1200 },
+    font: {
+      fontFiles: [resolve(DIR_ROOT, './public/SourceHanSansSC.otf')],
+      loadSystemFonts: false,
```

Comment:
```suggestion
      loadSystemFonts: true,
```

Should we toggle it on default?

Reply:
The official document show that if load system fonts that will cost more time